woocommerce/woocommerce-ios
# [Mobile Payments] Modal views

Closes #3980 

⚠️ This PR is against the feature branch implementing support for Card Present Payments, not against develop ⚠️
⚠️ Testing this PR requires a hardware reader ⚠️

This is a first iteration on the final layout of the modal views. I don't love this implementation, but we need to push something forward now in order to unblock our efforts to provide a better UI for discovering and pairing with cards.

## Changes
* Add a view controller that contains all the labels and buttons used by the different states of the modal view that are necessary to collect a payment. This might not be the best possible approach, but it is what we need at this point.
* This is responsible for styling and showing/hiding labels and buttons.
* Add different "viewmodels" that provide the content for the view controller to render, and encapsulate the actions associated to the buttons.

Again, this is not perfect, but it's better than nothing, and it moves the needle forward. We can start logging tickets for improvements right away. 😄 


https://user-images.githubusercontent.com/2722505/116465776-acc41300-a83b-11eb-9982-87558c475c7b.mov

| Light appearance |  Dark Appearance |
| ---- | ---- |
| <img src="https://user-images.githubusercontent.com/2722505/116466450-7dfa6c80-a83c-11eb-9162-e1c5edac0e92.PNG" width="350"/> | <img src="https://user-images.githubusercontent.com/2722505/116466447-7d61d600-a83c-11eb-90eb-8024676ef4b3.PNG" width="350"/> |
| <img src="https://user-images.githubusercontent.com/2722505/116466452-7e930300-a83c-11eb-9fc5-606d41614b7d.PNG" width="350"/> | <img src="https://user-images.githubusercontent.com/2722505/116466451-7dfa6c80-a83c-11eb-97ec-4fd0a2b95279.PNG" width="350"/> |
| <img src="https://user-images.githubusercontent.com/2722505/116466454-7e930300-a83c-11eb-925c-08c39af668b0.PNG" width="350"/> | <img src="https://user-images.githubusercontent.com/2722505/116466455-7f2b9980-a83c-11eb-93d9-6f3f619b63a8.PNG" width="350"/> |


## How to test
Testing requires:

1. Setting up a store for Card Present Payment Testing (also P91TBi-4BH-p2 for more specific instructions)
2. An external hardware reader.

* Checkout the branch, run bundle exec pod install, just for kicks and giggles.
* Build and run on a device.
* Switch an external card reader on. The only card reader supported by now is the BBPOS Chipper 2X BT.
* Log in to an account that matches a store that has been configured for testing.
* Navigate to Settings in the app. (Tap the gear button in the top right)
* In the "Store Settings" section, tap "Manage Card Reader".
* Make sure the card reader is on and the blue light is blinking (we don't do any error handling yet, so it's important that things are setup for the happy path)
* Tap "Connect card reader"
* Wait until the app suggests one reader to connect to.
* Tap "Connect to reader"
* Make sure the reader light turns solid blue.
* Navigate to an order eligible for Card Present Payments (it has to be a store enrolled to WCPay, and the order needs to have as payment method associated Cash on Delivery or none).
* Tap the "Collect Payment" button.
* Tap the card to the reader when requested.
* If everything goes well, the card should be charged, and the payment should be completed.
* At that point, tapping the "Print receipt" button will trigger a printout. The "Email receipt button" does nothing yet.

Update release notes:
